src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
             music.play();
+        }
+    });
+
+    events::sound_effects_event.connect([&](const events::AudioData &d) {
+        std::string filename = d.filename;
+
+        sounds[filename]->setVolume(d.volume);
+        sounds[filename]->setLoop(d.loop);
+
+        if (!muted) {
```

Comment:
Same reason as above (mostly for the case of continuous sound effects like footsteps). We can exit early for one-time sounds like:
` 
if (muted && d.loop) {
    // Change state to paused so that when audio is un-muted, the looped sound will play
    sounds[filename]->setVolume(d.volume);       
    sounds[filename]->setLoop(d.loop);     
    sounds[filename]->play();
    sounds[filename]->pause();
}
if (!muted) {
    sounds[filename]->setVolume(d.volume);
    sounds[filename]->play();
}
`
What do you think?